### PR TITLE
siamrpn: Single Object tracking 2d

### PR DESCRIPTION
**Pull request by vivinousi**
This PR adds SiamRPN, including a ROS1 node, custom dataset support and three demos

### CODE REVIEW

File: src/opendr/perception/object_tracking_2d/siamrpn/siamrpn_learner.py [hunk 1]
```diff
@@ -0,0 +1,569 @@
+# Copyright 2020-2022 OpenDR European Project
```
Comment (tsampazk) [inline src/opendr/perception/object_tracking_2d/siamrpn/siamrpn_learner.py:1]: As a general suggestion for the learner, the method order could be changed to somewhat match the order of the methods in the docs (ctor, fit, eval, infer, save, load, download) and/or maybe put the private/static and not implemented methods at the bottom, or right after to the method that uses them.

File: projects/opendr_ws/src/perception/scripts/object_tracking_2d_siamrpn.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
```
Comment (tsampazk) [inline projects/opendr_ws/src/perception/scripts/object_tracking_2d_siamrpn.py:1]: Could you please rename the node script to `object_tracking_2d_siamrpn_node.py`, as all the ros1 nodes will be renamed this way in #364. That PR will probably be merged before this, and the package name will also change from `perception` to `opendr_perception`.

Other than that the node looks good!

**Review by tsampazk — CHANGES_REQUESTED** [review]
Thanks @vivinousi for the new tool! Changes LGTM apart from some minor comments.

File: src/opendr/perception/object_tracking_2d/siamrpn/siamrpn_learner.py (hunk 1 above)
Comment (vivinousi) [reply src/opendr/perception/object_tracking_2d/siamrpn/siamrpn_learner.py:1]: resolved in [fa6eeb7](https://github.com/opendr-eu/opendr/pull/367/commits/fa6eeb7ee3ff2dd586bf8f351f3e666f430cc9e3)

**Review by tsampazk — DISMISSED** [review]
LGTM, thank you!

**Review by tsampazk — DISMISSED** [review]
I resolved the conflicts with develop and applied some minor fixes.

**Review by ad-daniel — APPROVED** [review]
Just did some minor adjustments and fixed the merge conflict, other than that it looks good me. Thank you!

**Review by tsampazk — APPROVED** [review]

**Pull request merged**